src/main/java/undo/actions/Action.java

```diff
@@ -0,0 +1,43 @@
+package undo.actions;
+
+public interface Action<T> {
+
+    @SuppressWarnings("SpellCheckingInspection")
+    enum ActionType { CHANGELABELS }
+
+    ActionType getType();
+
+    String getDescription();
+
+    T act(T t);
```

Comment:
Name `act` was chosen to follow some convention? If not, `do` goes nicely with `undo` :-)